portfolio/src/main/java/com/google/sps/data/BookServiceClient.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.data;
+ 
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.sps.data.Book;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.lang.NullPointerException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+ 
+// Given a book name, fetches book info json by calling Google books api and returns as string.
+public class BookServiceClient {
+
+  /**
+  * Validates the book name and gets the most relevent result
+  * @param bookName: the name of the book you want to search for
+  */
+  public static String getBookInfo(String bookName) throws Exception{
+    validate(bookName);
+    String allResults = queryBooksAPI(bookName);
+    return getTopResult(allResults);
+  }
+
+  /**
+  * Takes the response from the book api and returns only the top result
+  * @param bookApiResults: the json string given by the api
+  */
+  private static String getTopResult(String booksApiResults){
+    JSONObject jsonObject = new JSONObject(booksApiResults);
+    JSONArray allItems = jsonObject.getJSONArray("items");
+    JSONObject firstItem = allItems.getJSONObject(0);//returns top result'
+    return firstItem.toString();
+  }
+
+  /**
+  * Provided the bookname it encodeds and formats it with the api
+  * url and gets the results from the api by calling queryUrl 
+  * @param bookName: the name of the book you want to search for
+  */
+  private static String queryBooksAPI(String bookName) throws Exception{
+    String encodedBookName = null;
+    encodedBookName = URLEncoder.encode(bookName, "UTF-8");
+    String url = String.format("https://www.googleapis.com/books/v1/volumes?country=US&q=%s", encodedBookName);
+    try{
+        return queryURL(url);
+    }
+    catch(Exception e){
+        throw new Exception("Could not query URL",e);
+    }
```

Comment:
Catching exception doesn't seem useful here, you can just replace this with :

`return queryURL(url);`